### ISSUE TITLE
attr.ib(order) function key

### ISSUE THREAD

**Issue by graingert** [python-attrs/attrs#624]
eg:

```python
import attr
import natsort

_natsort_key = natsort.natsort_keygen()

class Ham:
    name = attr.ib(order=_natsort_key)
    id = attr.ib()
```

**Comment by gabbard** [comment]
Is the request here that if you have:

```python
import attr
import natsort

_natsort_key = natsort.natsort_keygen()

@attrs(cmp=True)
class Ham:
    name = attr.ib(order=_natsort_key)
    id = attr.ib()
```

then the generated comparison functions should treat that attribute as if `_natsort_key` had been appplied to it first?

**Comment by gabbard** [comment]
This seems potentially related to https://github.com/python-attrs/attrs/issues/435

**Comment by botant** [comment]
I agree that this seems like something that could be handled as part of #435. 

I'm working on it now, and I'll check if the proposed solution can accommodate this use case in a natural/intuitive way.

**Comment by hynek** [comment]
I believe this is fixed by #787. Please scream/re-open if not!
